unixfs/tar/reader.go

```diff
@@ -28,12 +29,11 @@ type Reader struct {
 	err        error
 }
 
-func NewReader(path path.Path, dag mdag.DAGService, resolver *path.Resolver, compression int) (*Reader, error) {
+func NewReader(path string, node *core.IpfsNode, compression int) (*Reader, error) {
```

Comment:
this should not take a node. it makes for a much smaller surface area, and a [lower privilege](http://en.wikipedia.org/wiki/Principle_of_least_privilege)